src/modules/UI/scenes/WalletList/components/XPubModal/XPubModal.ui.js

```diff
@@ -0,0 +1,79 @@
+// @flow
+
+import React, { Component } from 'react'
+import { Platform, Text, Alert, Clipboard } from 'react-native'
+import s from '../../../../../../locales/strings.js'
+// import StylizedModal from '../../../../components/Modal/Modal.ui'
+import { InteractiveModal } from '../../../../components/Modals/InteractiveModal/InteractiveModal.ui.js'
+import {PrimaryButton} from '../../../../components/Modals/components/PrimaryButton.ui.js'
+import { Icon } from '../../../../components/Icon/Icon.ui.js'
+import styles from './style.js'
+
+type XPubModalOwnProps = {
+
+}
+
+type XPubModalStateProps = {
+  xPubKeySyntax: string,
+  visibilityBoolean: boolean,
+}
+
+type XPubModalDispatchProps = {
+  onExit: () => void
+}
+
+type XPubModalState = {
+
+}
+
+type XPubModalComponentProps = XPubModalOwnProps & XPubModalStateProps & XPubModalDispatchProps
+
+export default class XPubModal extends Component<XPubModalComponentProps, XPubModalState> {
+  _onPressCopy = () => {
+    try {
+      Clipboard.setString(this.props.xPubKeySyntax)
+      Alert.alert(
+        s.strings.fragment_wallets_pubkey_copied_title,
+        s.strings.fragment_wallets_pubkey_copied_success,
+        [{
+          text: s.strings.string_ok, onPress: () => this.props.onExit()
+        }]
+      )
+    } catch (e) {
+      console.log('Error:', e.title, e.message)
+      Alert.alert('Error', s.strings.fragment_wallets_pubkey_copied_error)
+    }
+  }
+
+  render () {
+    const osPrefix = Platform.OS === 'ios' ? 'ios-' : 'md-'
+    // const iconNa = <IonIcon name={`${osPrefix}eye`} size={24} color={colors.primary} style={styles.icon} />
+    return (
+
+      <InteractiveModal isActive={this.props.visibilityBoolean} onBackButtonPress={this.props.onExit} onBackdropPress={this.props.onExit} onModalHide={this.props.onExit}>
+        <InteractiveModal.Icon>
+          <Icon style={{}} type={'ionIcons'} name={`${osPrefix}eye`} size={30} />
```

Comment:
Please use a cross platform icon

Reply:
This is not how the Icon component is set up to work.
you should 
1. import * as Constants from '../../../../../../constants'
2. the type should be = {Constants.ION_ICONS} // this is referenced in the IconConstants.js file in the top level constants directory 
3. add the constant to the IconConstants file export const EYE = 'ios-eye'
we keep references to all the icons in the constants file to keep them centralized. 
Also , there is no Android-eye icon in the actual icon library so it will not work on android.